Add documentation for Sass configuration options (#8587)
jekyll/jekyll: docs/_docs/configuration/sass.md

---
title: Sass/SCSS Options
permalink: "/docs/configuration/sass/"
---

> By default, Jekyll looks for Sass partials in the `_sass` directory relative to your site's `source` directory. You can change the default load path or specify additional load paths with the following options:

Reviewer: ```suggestion
By default, Jekyll will look for Sass partials in the `_sass` directory relative to your site's `source` directory. You can change the default load path or specify additional load paths with the following options:
```

> * **sass_dir** - A filesystem-path which should be searched for Sass partials.

Reviewer: "filesystem-path" is verbose. Just say directory.

```suggestion
- **sass_dir** - Look for Sass partials in this directory.
```

> * **load_paths** - An array of additional filesystem-paths which should be searched for Sass partials.

Reviewer: ```suggestion
* **load_paths** - Search for additional Sass partials in this array of directory paths.
```

<div class="note info">
  <p>

>     Note that filesystem-paths specified in the <code>sass_dir</code> and

Reviewer: ```suggestion
    Note that directory paths specified in the <code>sass_dir</code> and
```

    <code>load_paths</code> options are resolved relative to your site's
    <code>source</code>, not relative to the location of the
    <code>_config.yml</code> file.
  </p>
</div>

> The following additional Sass configuration options are offered by the [jekyll-sass-converter](https://github.com/jekyll/jekyll-sass-converter) plugin.

Reviewer: This is not core Jekyll - maybe leave this out and rather link to that plugin's docs?

When the plugin changes, these docs become out of date, so rather just provide a reference.

> * **style** - Sets the style of the CSS-output. Can be `nested`, `compact`, `compressed`, or `expanded`. See the [Sass docs](https://sass-lang.com/documentation/cli/dart-sass#style) for details.

Reviewer: Recommended markdown style is dashes for outer level and stars for one level down. And 4 spaces for bullet points rather than 2 (for readability and because 4 spaces handles things better like code blocks within bullet points.

Also `CSS output` instead of `CSS-output`.
e.g.

```md
- **style** - Sets the style of the CSS output. Can be `nested`, `compact`, `compressed`, or `expanded`. See the [Sass docs](https://sass-lang.com/documentation/cli/dart-sass#style) for details.
- **sourcemap** - Controls when source maps shall be generated.
    * `never` — causes no source maps to be generated at all.
```

Also, since you split `sourcemap` values on multiple lines, you could do that here.

```md
- **style** - Set the style of the CSS output. See the [Sass docs](https://sass-lang.com/documentation/cli/dart-sass#style) for details. Valid values:
    * `nested`
    * `compact`
    * `compressed`
    *  `expanded`.
```

> * **sourcemap** - Controls when source maps shall be generated.

Reviewer: ```suggestion
* **sourcemap** - Set this to control whether source maps will be generated.
```

>   * `never` — causes no source maps to be generated at all.

Reviewer: ```suggestion
  * `never` — Source maps are never generated.
```

>   * `always` — source maps will always be generated.

Reviewer: ```suggestion
  * `always` — Do not generated source maps.
```

>   * `development` — source maps will only be generated if the site is in development environment. That is, when the environment variable [`JEKYLL_ENV`]({{ '/docs/configuration/environments/' | relative_url }}) is set to `development`.

Reviewer: ```suggestion
  * `development` — Generated source maps only in development mode. i.e. When the environment variable [`JEKYLL_ENV`]({{ '/docs/configuration/environments/' | relative_url }}) is set to `development`.
```

* **line_comments** - When set to `true`, the line number and filename of the source is included in the compiled CSS-file. Useful for debugging when the source map is not available, but might considerably increase the size of the generated CSS files.

> ### Default Configuration

Reviewer: ```suggestion
### Default Configuration

```
